src/Matrix.jl

```diff
@@ -845,7 +845,7 @@ function -(x::MatrixElem{T}, y::MatrixElem{T}) where {T <: NCRingElement}
    return r
 end
 
-function *(x::MatElem{T}, y::MatElem{T}) where {T <: NCRingElement}
+function *(x::T, y::T) where {T <: MatrixElem}
```

Comment:
@JohnAAbbott ... while this was restricted to the "both arguments have the same type.

But yeah it really should be
```suggestion
function *(x::T, y::T) where {T <: MatElem}
```
and then there should be `MatRingElem` methods for `*`, `+`, `-` etc. which delegate to the the underlying "plain" matrix.

I.e., also the existing changes in this PR in lines 826 and 837 of this file need to be furhter modified.